### PR TITLE
feat(pr-review): raise patch limit to 25k and extend Contents API fallback to added files

### LINKED ISSUE #1384: feat(pr-review): replace hard patch truncation with summarization pre-pass for oversized files
## Background

PR #1383 fixed the immediate bug: patches exceeding `max_patch_chars_per_file` were silently dropped with no fallback. The fix truncates to the char limit with an `[APTU: ...]` annotation.

This is a correctness fix, not a quality fix. The current approach has two remaining problems:

1. **The default limit (10,000 chars) is too low for real-world PRs.** A 347-line Python file runs ~11,879 chars; a typical 400-line Rust module with comments runs 12-25k chars. Most non-trivial added or modified files exceed the limit.

2. **Hard truncation is architecturally inferior to summarization.** Cutting at N chars mid-function gives the AI partial, context-broken content. The better tools (CodeRabbit, GitHub Copilot PR review) use a summarization pre-pass: a cheap/fast model compresses oversized files down to the functions touched by the change, so the frontier model sees a brief rather than a raw truncated patch.

## What enterprise tools do

- **CodeRabbit:** cheap-model (GPT-4.1 nano/mini) compression pass on files exceeding context -- a 4,000-line file becomes only the functions the change touches. Recursive summarization across commits for large PRs (100+ files). Truncation is a last resort, not the designed behavior.
- **GitHub Copilot PR review:** agentic task graph -- a planner decomposes large PRs into sub-tasks rather than feeding the whole diff to a single model call. No hard char-limit truncation.

### 1. Raise the default `max_patch_chars_per_file`

Raise from 10,000 to ~25,000 chars in `ReviewConfig`. This eliminates most false truncations with no code change. The model context window is not the bottleneck at this scale.

### 2. For added files: use Contents API fallback instead of diff truncation

Added files (status `added`) have no meaningful diff hunk -- the entire file is new. When the patch exceeds budget for an added file, fetch the full file content via the Contents API (the fallback already implemented for GitHub-truncated patches in `github/pulls.rs:328-388`) and include it as file content rather than a truncated diff. This is already implemented for `patch_truncated` files; extend the trigger condition to include size-budget drops for added files.

### 3. Long-term: summarization pre-pass for oversized non-added files

When a modified file's patch exceeds `max_patch_chars_per_file` after the limit raise, generate a compact summary of the diff (changed functions, key logic changes, removed/added imports) using a fast/cheap model call, and include that summary in the prompt instead of the raw truncated patch. This mirrors CodeRabbit's architecture.

Implementation should be gated on a config flag (`[review] summarize_oversized_patches = true`, default off) until the behavior is validated.

## Scope

- `crates/aptu-core/src/ai/review_context.rs` -- `ReviewConfig` default for `max_patch_chars_per_file`
- `crates/aptu-core/src/ai/prompts/mod.rs` -- added-file detection + summarization pre-pass
- `crates/aptu-core/src/github/pulls.rs` -- extend Contents API fallback trigger

## References

- Fixes the root architectural gap identified in #1382
- PR #1383 (immediate fix, now merged)
- CodeRabbit architecture: https://medium.com/data-science-collective/how-coderabbit-actually-works-331aeab55ec8

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Raises the default `max_patch_chars_per_file` from 10,000 to 25,000 characters and extends the existing Contents API fallback to cover Added/Renamed/Copied files whose patches exceed the per-file character budget. This allows the AI to review full file contents for newly added files even when the diff patch is too large, improving code review quality for substantial feature additions.

Closes #1384

## Changes

- `crates/aptu-core/src/config/review.rs`: Updated `max_patch_chars_per_file` default from 10,000 to 25,000 and added test validating the new default
- `crates/aptu-core/src/github/pulls.rs`: Extended Contents API fallback block in `fetch_pr_details` to fetch full file content for Added/Renamed/Copied files with oversized patches; includes case-insensitive status matching, dedup guard for already-populated full content, and proper error handling with logging; extracted repeated `pr.head` SHA expression to a local variable and added inline comment explaining the `to_lowercase()` necessity

## Test plan

- [x] Tests pass: 484 passed, 0 failed
- [x] Linter clean: `cargo clippy -- -D warnings`
- [x] Security scan clean: `aptu scan-security` returned zero findings
- [x] New tests cover: happy path (status=Added + oversized patch), edge cases (Modified status excluded, dedup guard prevents re-fetch, Contents API failures logged), and default value verification

## Implementation details

- Case-insensitive status matching (`added|renamed|copied`) ensures robustness against Octocrab's `Debug`-formatted enum output which produces mixed-case values
- Dedup guard (`full_content.is_none()`) prevents redundant API calls if Contents API was already invoked by the earlier `is_patch_truncated` block
- Fallback block runs before `drop_patches_by_size` so the AI can use the full content in the prompt
- Error handling logs warnings but preserves the original patch so `drop_patches_by_size` can handle it as before (safety valve)
- Repeated `pr.head.as_deref().map(|h| h.sha.as_str()).unwrap_or_default()` expression extracted to a local `head_sha` binding shared by both fallback loops

## Breaking changes

None. This is a configuration default change and enhancement to existing fallback logic. No API changes or behavior modifications to the public interface.

### CODE REVIEW

File: crates/aptu-core/src/github/pulls.rs [hunk 1]
```diff
@@ -185,6 +185,52 @@ pub async fn fetch_pr_details(
...
+        );
+        let patch_too_large =
+            file.patch.as_deref().map_or(0, str::len) > review_config.max_patch_chars_per_file;
+        if is_added_renamed_copied && patch_too_large && file.full_content.is_none() {
+            match fetch_file_contents_single(
+                client,
+                owner,
+                repo,
+                &file.filename,
+                pr.head
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/github/pulls.rs]: Fixed in the latest commit. The repeated expression is now extracted to a local  variable.

File: crates/aptu-core/src/github/pulls.rs [hunk 2]
```diff
@@ -185,6 +185,52 @@ pub async fn fetch_pr_details(
...
+    for file in &mut pr_files {
+        let is_added_renamed_copied = matches!(
+            file.status.to_lowercase().as_str(),
+            "added" | "renamed" | "copied"
+        );
+        let patch_too_large =
+            file.patch.as_deref().map_or(0, str::len) > review_config.max_patch_chars_per_file;
+        if is_added_renamed_copied && patch_too_large && file.full_content.is_none() {
+            match fetch_file_contents_single(
+                client,
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/github/pulls.rs:204]: Acknowledged. Added an inline comment explaining that format!("{:?}", f.status) yields mixed-case values, so to_lowercase() is necessary for case-insensitive matching.

File: crates/aptu-core/src/github/pulls.rs [hunk 3]
```diff
@@ -185,6 +185,52 @@ pub async fn fetch_pr_details(
         }
     }
 
+    // Contents API fallback for Added/Renamed/Copied files with oversized patches.
+    // Fetch full content from Contents API so the AI can review the full file, even though
+    // the patch exceeds the character budget.
+    for file in &mut pr_files {
+        let is_added_renamed_copied = matches!(
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/github/pulls.rs:196]: Acknowledged. Added an inline comment above the to_lowercase() call explaining the rationale.

File: crates/aptu-core/src/github/pulls.rs [hunk 4]
```diff
@@ -185,6 +188,50 @@ pub async fn fetch_pr_details(
         }
     }
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/github/pulls.rs:190]: Acknowledged. The inline `matches!` expression is self-contained and only used in one place; extracting a method on `PrFile` would add indirection without a concrete reuse benefit at this stage. If a second call site appears this would be the right refactor to make.
